### PR TITLE
2.0.0 alpha

### PR DESCRIPTION
**Pull request by poteto**
- [ ] Compat
- [x] Remove injetion factories
- [x] Use contextual components
- [x] Closing a flash message is now an action emitted instead of a
  default behaviour on click
- [x] Remove feature to output class names for Bootstrap / Foundation -
  consuming app can simply use a Helper to generate those
- [x] Update ember-cli
- [x] Update readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -233,71 +205,33 @@ See the [options](#options) section for information about flash message specific
...
-### Custom flash message component
-If the provided component isn't to your liking, you can easily create your own. All you need to do is pass in the `flash` object to that component:
-
-```handlebars
-{{#each flashMessages.queue as |flash|}}
-  {{custom-component flash=flash}}
-{{/each}}
+```hbs
+{{#flash-messages flashes=flashMessages.arrangedQueue as |flash|}}
+  {{#flash.component class=(concat "alert alert-" flash.type) click=(action flash.close) as |flashMessage|}}
```
Comment (adrind) [inline README.md]: Foundation 6 merged alerts and panels into a 'callout' class, is 2.0 going to provide this support or should we handle it with our own component?

http://foundation.zurb.com/sites/docs/callout.html

File: README.md [hunk 2]
```diff
@@ -1,64 +1,52 @@
...
 [![npm version](https://badge.fury.io/js/ember-cli-flash.svg)](http://badge.fury.io/js/ember-cli-flash) [![Build Status](https://travis-ci.org/poteto/ember-cli-flash.svg)](https://travis-ci.org/poteto/ember-cli-flash) [![Ember Observer Score](http://emberobserver.com/badges/ember-cli-flash.svg)](http://emberobserver.com/addons/ember-cli-flash) [![Code Climate](https://codeclimate.com/github/poteto/ember-cli-flash/badges/gpa.svg)](https://codeclimate.com/github/poteto/ember-cli-flash) 
 
 This `ember-cli` addon adds a simple flash message service and component to your app. Just inject with `Ember.inject.service` and you're good to go!
 
+## Warning
+
+This is the readme for the `2.0.0-alpha` series, some information might be incorrect or outdated. For prior versions, please refer to [previous tags](https://github.com/poteto/ember-cli-flash/tags). Due to the use of contextual components, the `2.x` series of ember-cli-flash will require the use of the `ember-hash-helper-polyfill` addon if you are on Ember < 2.3.
+
 ## Installation
 You can install either with `ember install`:
```
Comment (GCheung55) [inline README.md]: There's only one way to install now, so `either` should be removed.

**Comment by keradus** [comment]
Nice work so far !
Is there any ETA for releasing 2.0 ?

**Comment by jrowlingson** [comment]
@poteto I would suggest adding something like https://github.com/DockYard/ember-composable-helpers/pull/196 to the list as it is a breaking change

**Comment by danielruediger** [comment]
@poteto is there any ETA for releasing 2.0❓

File: package.json [hunk 3]
```diff
@@ -26,13 +26,14 @@
   "license": "MIT",
   "dependencies": {
     "ember-cli-babel": "^5.1.6",
-    "ember-new-computed": "^1.0.3",
-    "ember-cli-htmlbars": "^1.0.3"
+    "ember-cli-htmlbars": "^1.0.3",
+    "ember-hash-helper-polyfill": "0.1.0",
```
Comment (mike-north) [inline package.json]: May not be necessary given the range of ember versions you're testing against

File: tests/unit/components/flash-message-test.js [hunk 4]
```diff
@@ -35,30 +35,28 @@ moduleForComponent('flash-message', 'FlashMessageComponent', {
...
 });
 
-test('exiting the flash object sets exiting on the component', function(assert) {
+test('exiting the flash object sets `isExiting` on the component', function(assert) {
   assert.expect(2);
 
   const component = this.subject({ flash });
   this.render();
-  assert.equal(get(component, 'exiting'), false, 'it initializes with `exiting` set to false');
+  assert.equal(get(component, 'isExiting'), false, 'it initializes with `isExiting` set to false');
```
Comment (sbatson5) [inline tests/unit/components/flash-message-test.js]: Thoughts on just `assert.notOk` since we are testing booleans?

**Comment by poteto** [comment]
@sbatson5 Do you have any spare cycles to take this over the line? I think it is pretty close and is going to be a superior experience compared to what we have now

**Comment by sbatson5** [comment]
@poteto I should have some time in the next couple of weeks to try and take this over

**Pull request closed without merging**